### PR TITLE
Adding AsPublicInterfaceImplementation ability to handle expression-bodied

### LINKED ISSUE #65931: `AsPublicInterfaceImplementation` doesn't handle expression-bodied correctly.
```csharp
public interface IGeneral
{
    object DoSomething();
}

public class C : IGeneral
{
    object IGeneral.DoSomething() => new();
}
```

Call `AsPublicInterfaceImplementation` on the explicit implementation above. It produces:

image

The following code should check for ExpressionBody

https://github.com/dotnet/roslyn/blob/683e22d4a2351e5495e18fbb2f7089f2f1a9679d/src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs#L581-L619

Accessors should be fixed too.

### PR DESCRIPTION
**Pull request by kinkeadian25**
Closes #65931

### CODE REVIEW

**Review by Youssef1313 — APPROVED** [review]
LGTM

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 1]
```diff
@@ -592,15 +594,15 @@ private static SyntaxNode WithBodies(SyntaxNode declaration)
...
 
                 case SyntaxKind.OperatorDeclaration:
                     var op = (OperatorDeclarationSyntax)declaration;
-                    return (op.Body == null) ? op.WithSemicolonToken(default).WithBody(CreateBlock()) : op;
+                    return (op.Body == null && op.ExpressionBody == null) ? op.WithSemicolonToken(default).WithBody(CreateBlock()) : op;
 
                 case SyntaxKind.ConversionOperatorDeclaration:
                     var cop = (ConversionOperatorDeclarationSyntax)declaration;
-                    return (cop.Body == null) ? cop.WithSemicolonToken(default).WithBody(CreateBlock()) : cop;
+                    return (cop.Body == null && cop.ExpressionBody == null) ? cop.WithSemicolonToken(default).WithBody(CreateBlock()) : cop;
```
Comment (Youssef1313) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:605]: Could these be re-written as follows?

```csharp
                case SyntaxKind.MethodDeclaration:
                case SyntaxKind.ConversionOperatorDeclaration:
                case SyntaxKind.OperatorDeclaration:
                    var method = (BaseMethodDeclarationSyntax)declaration;
                    return (method.Body == null && method.ExpressionBody == null) ? method.WithSemicolonToken(default).WithBody(CreateBlock()) : method;
```

Comment (kinkeadian25) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs]: Cool suggestion! Thanks!

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks Ian!

**Comment by kinkeadian25** [comment]
Thanks so much @CyrusNajmabadi and @Youssef1313

**Pull request merged**